buzzerComponent/buzzer.c

```diff
@@ -11,278 +25,289 @@
-static int period = 0;
+// The on percentage of the buzzer on/off duty cycle (0 to 100).
+static uint DutyCycleOnPercent = 50;
+
+// The total number of milliseconds in the full duty cycle period (on + off).
+// Must be >= 100 and <= 3600000 (i.e. 1 hour).
+static uint PeriodMs = 2000;
 
 // The timer used to run the duty cycle
 static le_timer_Ref_t Timer = NULL;
-static uint32_t dc_on_timer = 0;
-static uint32_t dc_off_timer = 0;
-static bool buzzer_on = false;
 
-static void SetBuzzer(void)
+// true if the buzzer is currently on (buzzing).
+static bool BuzzerOn = false;
+
+//--------------------------------------------------------------------------------------------------
+/**
+ * Makes the buzzer sound at a given frequency in Hz.
+ */
+//--------------------------------------------------------------------------------------------------
+static void SetBuzzerHz
+(
+    uint freq    ///< The buzzer frequency, in Hz. Use BUZZER_OFF_FREQ to stop the buzzer.
+)
 {
-    if (freqFp == NULL)
+    // Path to the RTC CLKOUT control file in sysfs.
+    static const char BuzzerFreqPath[] = "/sys/bus/i2c/drivers/rtc-pcf85063/8-0051/clkout_freq";
+
+    static FILE *FreqFile = NULL;
+
+    if (FreqFile == NULL)
     {
-        freqFp = fopen(BuzzerFreqPath, "r+");
-        if (freqFp == NULL)
+        FreqFile = fopen(BuzzerFreqPath, "r+");
+        if (FreqFile == NULL)
         {
-            LE_ERROR("Open clkout sysfs file('%s') failed(%d)", BuzzerFreqPath, errno);
-            return;
+            LE_FATAL("Opening file (%s) failed (%m)", BuzzerFreqPath);
         }
     }
 
-    LE_DEBUG("Clkout frequency %d Hz", current_freq);
-
-    if (fprintf(freqFp, "%d", current_freq) == -1)
+    if (fprintf(FreqFile, "%d", freq) == -1)
     {
-        LE_ERROR("Write clkout file('%s') failed (%d)", BuzzerFreqPath, errno);
+        LE_FATAL("Write to file (%s) failed (%m)", BuzzerFreqPath);
     }
 
-    if (fflush(freqFp) == -1)
+    if (fflush(FreqFile) != 0)
     {
-        LE_ERROR("fflush of /sys/bus/i2c/drivers/rtc-pcf85063/8-0051/clkout_freq failed");
+        LE_FATAL("fflush of file (%s) failed (%m)", BuzzerFreqPath);
     }
 }
 
-static void DcHandler(le_timer_Ref_t DcTimerRef)
+//--------------------------------------------------------------------------------------------------
+/**
+ * Start at the beginning of a duty cycle.
+ */
+//--------------------------------------------------------------------------------------------------
+static void StartCycle()
+{
+    SetBuzzerHz(Frequency);
+    BuzzerOn = true;
+    le_timer_SetMsInterval(Timer, PeriodMs * DutyCycleOnPercent / 100);
+    le_timer_Start(Timer);
+}
+
+//--------------------------------------------------------------------------------------------------
+/**
+ * Disable the buzzer, immediately stopping it, even if it's in the middle of a duty cycle.
+ */
+//--------------------------------------------------------------------------------------------------
+static void StopCycle()
```

Comment:
I prefer that all zero argument functions explicitly specify `void` as their argument list.  Technically there's no difference in the case of function definitions, but often times definitions get copied into declarations later on.

Reply:
I tend to agree.  I'll change this.